collectors/node.go

```diff
@@ -219,6 +226,14 @@ func (nc *nodeCollector) collectNode(ch chan<- prometheus.Metric, n v1.Node) {
 
 	addGauge(descNodeSpecUnschedulable, boolFloat64(n.Spec.Unschedulable))
 
+	// Collect node taints
+	for _, taint := range n.Spec.Taints {
+		// Taints are applied to repel pods from nodes that do not have a corresponding
+		// toleration.  Many node conditions are optionally reflected as taints
+		// by the node controller in order to simplify scheduling constraints.
+		addGauge(descNodeSpecTaint, 1, taint.Key, taint.Value, string(taint.Effect))
```

Comment:
I'm a bit worried about the cardinality here. Do you have some numbers of combinations that are typical in clusters for each node?

Reply:
Good point; given normal users likely don't have authorization to mutate taints on a node, it does seem unlikely that these would grow unbounded in practice, but as a defensive measure would it make any sense to introduce a hard coded upper bound to the amount of taints which we'll process (and produce an event or log if that limit is reached)?

Reply:
i anticipate an upper bound of num(taints) < 10 for the most tainted of nodes.

~5 node conditions -> mapped to node taints if you enable a current alpha flag which syncs them
~1 node taint identified now for volumes stuck
~1 node taint to steer workloads with matching toleration (i.e. accelerated, storage-optimized, highmem-lowcpu, etc.)

cluster-operators or sufficiently privileged actors will apply taints to nodes.